src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -122,10 +124,18 @@ var ReactCompositeComponentMixin = {
    * @final
    * @internal
    */
-  mountComponent: function(rootID, transaction, context) {
+  mountComponent: function(
+    rootID,
+    transaction,
+    nativeParent,
+    nativeContainerInfo,
+    context
+  ) {
     this._context = context;
     this._mountOrder = nextMountID++;
     this._rootNodeID = rootID;
+    this._nativeParent = nativeParent;
+    this._nativeContainerInfo = nativeContainerInfo;
```

Comment:
The top-level wrapper is hacky but that's sort of solving a different problem. We need to store this stuff because we need it for updates. I guess for the root case we could trigger updates in a different way that passes this down again.